src/OpenTelemetry/Metrics/MeterProviderBuilderBase.cs

```diff
@@ -32,6 +32,10 @@ public abstract class MeterProviderBuilderBase : MeterProviderBuilder
     {
         internal const int MaxMetricsDefault = 1000;
         internal const int MaxMetricPointsPerMetricDefault = 2000;
+
+        // Builds noop MeterProvider when this flag is set.
+        // This flag is controlled through reflection.
+        private static bool disableMeterProviderBuilder = false;
```

Comment:
Is there a scenario where metrics/traces need to be disabled independently?
What is the contract and how to maintain the contract?

Reply:
Maybe consider something similar to https://github.com/open-telemetry/opentelemetry-dotnet/blob/6b7f2dd77cf9d37260a853fcc95f7b77e296065d/src/OpenTelemetry/SuppressInstrumentationScope.cs#L25-L29.